### DISCUSSION TITLE
device.get("    ") parameters

### DISCUSSION THREAD

**Discussion by NickP-tfw** [CrowdStrike/falconpy#742] [Show and Tell]
Hi, I've just started playing around with this. Its working :-)

I'm using "sensor_versions_by_hostname.py" to play around by editing little bits to see the effects.
I can see how, on line 69, device.get("agent_version", None) works

I've looked at the Hosts collection page and can substitute "os_version" "bios_version" to get results but I specifically want the OS Build eg for Windows 10 it might be "19044".  

I've just solved my own problem by playing with falcon.get_device_details. This is "os_build" its not on the hosts collection page, at least I couldn't find it. You might want to add it.

Thanks
Nick

**Comment by crowdstrikedcs** [comment]
Hey There @NickP-tfw !

Good suggestion, that sample is specifically for returning the Sensor Version & Hostname pairings in the environment.

We are building a dictionary of attributes [here](https://github.com/CrowdStrike/falconpy/blob/1e2467f194d80f46e9db5498bbc7050c11a01606/samples/hosts/sensor_versions_by_hostname.py#L67) with `device.hostname` and `device.agent_version` if we added `device.os_build` here additionally we could then include its value [here](https://github.com/CrowdStrike/falconpy/blob/1e2467f194d80f46e9db5498bbc7050c11a01606/samples/hosts/sensor_versions_by_hostname.py#L104) to have it printed. Something like this

`print(f"{DISPLAYED}: {detail['hostname']} is on version {detail['agent_version']} OS Build: {detail['os_build']}")`

Happy to have you submit a new sample to the repo for this!

**Reply by NickP-tfw** [reply]
Happy to
I've already got it to list hosts that are #1 Windows 10 but not running 19044, #2 not on our internal network 
by hostname, make/model/serial and last seen with internal and external IPs to give me an idea of how they are working.
LP-103227           ,LENOVO,20VJS22800,PW019S58,Windows 10,19043,28-06-2022,15:19:50,169.254.999.120,90.999.0.216
LP-103244           ,LENOVO,20VJS22800,PW01KD7D,Windows 10,19043,29-07-2022,05:08:57,192.168.999.1,213.205.999.28
LP-103250           ,LENOVO,20VJS22800,PW019S5X,Windows 10,19043,29-07-2022,09:27:19,10.999.10.202,185.104.999.17

I think i need to work on the formatting :-)
